.github/workflows/on_target.yml

```diff
@@ -72,15 +29,29 @@ jobs:
       - name: Download artifact
         uses: actions/download-artifact@v4
         with:
-          name: oob-t91x-hex
+          name: firmware
           path: thingy91x-oob/tests/on_target/artifacts
 
+      - name: Set version
+        shell: bash
+        run: |
+          if [[ "${{ startsWith(github.ref, 'refs/tags/') }}" == "true" ]]; then
+            echo "VERSION=${{ github.ref_name }}" >> $GITHUB_ENV
+          else
+            echo "VERSION=${{ github.sha }}" >> $GITHUB_ENV
+          fi
+
       - name: Verify artifact path
         working-directory: thingy91x-oob
         run: |
           ls -l tests/on_target/artifacts
 
+      - name: Install dependencies
+        working-directory: thingy91x-oob/tests/on_target
+        run: |
+          pip install -r requirements.txt --break-system-packages
+
       - name: Run tests
-        working-directory: thingy91x-oob
+        working-directory: thingy91x-oob/tests/on_target
         run: |
-          ./tests/on_target/run-tests.sh
+          pytest -v --firmware-hex tests/on_target/artifacts/hello.nrfcloud.com-${{ env.VERSION }}-thingy91x-debug-app.hex
```

Comment:
change to `pytest -v --firmware-hex artifacts/hello.nrfcloud.com-${{ env.VERSION }}-thingy91x-debug-app.hex`